c++/src/kj/compat/http.c++

```diff
@@ -2339,10 +2336,13 @@ private:
   uint64_t length;
 
   kj::Promise<void> maybeFinishAfter(kj::Promise<void> promise) {
-    if (length == 0) {
-      return promise.then([this]() { doneWriting(); });
-    } else {
-      return kj::mv(promise);
+    // `length` shouldn't be mutated while we await `promise` ... but just in case, let's save it.
```

Comment:
If the length could change during the promise, then it would actually be more correct in this case to check it after, not before. But in fact, it can't change, hence the old code checked it before out of convenience. In any case, storing a copy here definitely isn't needed.